### PR TITLE
[4.x] Prevent logout flow being triggert when incorrect credentials are used

### PR DESCRIPTION
**Pull request by indykoning**
This PR prevents an incorrect password from clearing your cart during login.
By default when triggering login we do not want to clear carts or redirect to login

5.x: https://github.com/rapidez/core/pull/1257

### CODE REVIEW

**Comment by royduin** [comment]
Maybe we need some tests for this as it's such an edge case but really annoying for customers?

File: tests/playwright/checkout.spec.js [hunk 1]
```diff
@@ -39,3 +39,31 @@ checkoutTypes.forEach((type) => {
...
+
+    // Login
+    await productPage.addToCart(process.env.PRODUCT_URL_SIMPLE)
+    await checkoutPage.gotoCheckout()
+    await checkoutPage.login(email, password + '!')
+    await this.page.getByTestId('continue').click()
+    await this.page.waitForTimeout(500)
+    await this.page.waitForLoadState('networkidle')
+    await expect(page.getByTestId('notifications')).toHaveText(
+        'The account sign-in was incorrect or your account is disabled temporarily. Please wait and try again later.',
```
[the remark this answers is not in the document]
Comment (indykoning) [reply tests/playwright/checkout.spec.js:67]: This is the default Magento message, it can be changed via magento translations but we do not do that in the magento sample data container

**Comment by indykoning** [comment]
This will also be ready to merge once the screenshots are updated.
